src/utils.ts

```diff
@@ -2458,37 +2655,36 @@ export class newAxis extends EventEmitter {
 
   constructor(
     vector: any[],
-    public freeSpace: number,
+    public boundingBox: newRect,
     public origin: Vertex,
     public end: Vertex,
     public name: string = '',
     public initScale: Vertex,
-    private _nTicks: number = 10
+    protected _nTicks: number = 10
     ) {
       super();
       this.isDiscrete = vector === undefined ? true : typeof vector[0] == 'string';
       if (this.isDiscrete) {
-        if (vector) this.labels = newAxis.uniqueValues(vector)
-        else this.labels = ["0", "1"];
+        this.labels = vector === undefined ? ["0", "1"] : newAxis.uniqueValues(vector)
```

Comment:
Thanks, but since the current behavior was corrupted, i had to make some changes.
The result is not what you propose but everything is done with null instead of undefined